image_match/elasticsearch_driver.py

```diff
@@ -49,10 +49,11 @@ def search_single_record(self, rec):
         path = rec.pop('path')
         signature = rec.pop('signature')
 
-        fields = ['path', 'signature']
+        fields = ['path', 'signature', 'metadata']
 
         # build the 'should' list
-        should = [{'term': {word: rec[word]}} for word in rec]
+        should = [{'term': {word: rec[word]}} if rec[word]  else None for word in rec]
```

Comment:
@smallevil (I changed the name from `ext` to `metadata` for clarity): I need to think about these lines -- the user should have some flexibility in defining the metadata structure, so it's hard to say how exactly how it should be searched.  They may be stuck writing their own queries.

Let me think about it. Otherwise, aside from a couple minor changes, everything looks good.